### DISCUSSION TITLE
Is raw websocket supported?

### DISCUSSION THREAD

**Discussion by psnehanshu** [supabase/supabase#21204] [Questions]
Is there a way to implement raw websocket with Supabase? Realtime won't work because it has its own protocol. I want to implement [another protocol](https://en.m.wikipedia.org/wiki/Open_Charge_Point_Protocol), and thus it requires raw websocket. Is there a way to implement this using Edge functions?

**Comment by GaryAustin1** [comment]
The only websockets used by Supabase are on the multi-tenant realtime servers.  The realtime server connects to the Supabase database infrastructure with a direct database port.
  
All user connections to the database are thru direct database ports or the REST API provided by PostgREST over https.

You probably should look to an intermediate websocket to postgres database port server.  I see a lot of hits, but know nothing about them.
